src/script/backup/BackupRepository.ts

```diff
@@ -187,28 +199,132 @@ export class BackupRepository {
       const fileName = `${tableName}.json`;
       files[fileName] = encodedData;
     }
-
     files[Filename.METADATA] = encodedMetadata;
 
+    if (password) {
+      // encode header
+      const backupCoder = new BackUpHeader(user.id, password);
+      const backupHeader = await this.generateBackupHeader(user, password, backupCoder).catch(error => {
+        throw new Error('Backup error:', error);
+      });
+
+      // Encrypt the ZIP archive using the provided password
+      const {decodedHeader} = backupCoder.readBackupHeader(backupHeader);
+      const chaCha20Key = await backupCoder.generateChaCha20Key(decodedHeader);
+      const array = await this.worker.post<Uint8Array>({type: 'zip', files, encrytionKey: chaCha20Key});
+
+      // Prepend the combinedBytes to the ZIP archive data
+      const combinedArray = this.concatenateByteArrays([backupHeader, array]);
+      return new Blob([combinedArray], {type: 'application/zip'});
+    }
```

Comment:
Let's maybe create a separate method for creating encrypted backup when password is provided, this method will become much cleaner.

Reply:
i have thought about it. So i created smaller functions to make it more readable. if you read step by step its pretty clear. But i can create a separate function to improve the readability a bit more.